source/cppexpose/source/scripting/DuktapeScriptBackend.cpp

```diff
@@ -210,22 +203,33 @@ Variant DuktapeScriptBackend::fromDukStack(duk_idx_t index)
     else if (duk_is_object(m_context, index))
     {
         VariantMap map;
+        Object * object = nullptr;
 
         duk_enum(m_context, index, 0);
         while (duk_next(m_context, -1, 1))
         {
-            // Prevent the pointer to the C++ object that is stored in the Ecmascript object from being serialized
-            if (!(duk_is_pointer(m_context, -1) && fromDukStack(-2).value<std::string>() == s_duktapeObjectPointerKey))
+            // If a property s_duktapeObjectPointerKey exists, the object is a cppexpose::Object.
+            // In this case, extract the pointer and return that below
+            // Otherwise, continue to build a key-value map of the object's properties
+            if (duk_is_pointer(m_context, -1) && fromDukStack(-2).value<std::string>() == s_duktapeObjectPointerKey)
```

Comment:
Can we avoid the while loop altogether here (analog to the Qt implementation in gloperate)?

Reply:
I added an alternative implementation.